File: 19-budgets/npm-package/budgets.ts
import axios from 'axios';
const API = 'https://cost-katana-backend.store/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function createBudget(name: string, limit: number, period: string) {
  const res = await axios.post(`${API}/budget`, {name, limit, period},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Budget created:', res.data.data.id);
  return res.data.data;
}

export async function listBudgets() {
  const res = await axios.get(`${API}/budget`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log(`💰 Found ${res.data.data.length} budgets`);
  return res.data.data;
}

File: 19-budgets/python-sdk/budgets.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def create_budget(name, limit, period):
    res = requests.post(f"{API}/budget",
        json={"name": name, "limit": limit, "period": period},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"✅ Budget created: {res.json()['data']['id']}")
    return res.json()["data"]

def list_budgets():
    res = requests.get(f"{API}/budget",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"💰 Found {len(res.json()['data'])} budgets")
    return res.json()["data"]
